test/unit/argument_parser/format_parse_test.cpp

```diff
@@ -641,6 +641,26 @@ TEST(parse_type_test, parse_error_enum_option)
     EXPECT_THROW(parser.parse(), seqan3::argument_parser_error);
 }
 
+TEST(parse_test, issue2464_enum_user_input_error)
+{
+    // see issue https://github.com/seqan/seqan3/issues/2464
+    Other::bar option_value{};
+
+    {
+        const char * argv[] = {"./argument_parser_test", "-m", "9"};
+        seqan3::argument_parser parser{"test_parser", 3, argv, seqan3::update_notifications::off};
+        parser.add_option(option_value, 'e', "enum-option", "this is an enum option.");
+        EXPECT_THROW(parser.parse(), seqan3::argument_parser_error);
```

Comment:
```suggestion
        EXPECT_THROW(parser.parse(), seqan3::user_input_error);
```

Reply:
I tried user_input_error, but it is not thrown.. I dont understand why..

Reply:
Can you check what is thrown?
You can comment this line out.
Then just put `parser.parse();` on a new line.
Run the test with `--gtest_catch_exceptions=0` to prevent gtest from catching the exception.

(In general, `some_test_name --help` shows some sometimes useful googletest options)

Reply:
Might well be that it is actually an `argument_parser_error` or some other error that inherits from there, because the `user_input_error` is caught and rethrown somewhere :)